sdcm/cluster.py

```diff
@@ -3375,11 +3375,11 @@ def node_setup(self, node, verbose=False, timeout=3600):
         else:
             self._reuse_cluster_setup(node)
 
-        node.wait_db_up(verbose=verbose, timeout=timeout)
-        nodes_status = node.get_nodes_status()
-        check_nodes_status(nodes_status=nodes_status, current_node=node)
-
-        self.clean_replacement_node_ip(node)
+        if wait_db_up:
+            node.wait_db_up(verbose=verbose, timeout=timeout)
+            nodes_status = node.get_nodes_status()
+            check_nodes_status(nodes_status=nodes_status, current_node=node)
+            self.clean_replacement_node_ip(node)
```

Comment:
`clean_replacement_node_ip` is not related to waiting for DB up. It should be executed always or create a new param that controls it

Reply:
it is not related to wait_db_up, but this is called inside the function `node_setup` and we don't want to "clean" it until we know it is done, to avoid removing the `replacing_ip` before it even started, and we will miss the function to replace the node that was removed

Reply:
fair enough, but what bothers me is that when somebody sees a call `node_setup(wait_db_up=True)` without looking on implementation one will think that the method waits for DB up, but it is not true, in addition, it also cleans `replace_address_first_boot` parameter in `scylla.yaml`. Another thing when somebody will run it with wait_db_up=False, one needs to remember to run `clean_replacement_node_ip`. 
This is a bad API, but currently, I don't have a better solution except for adding a new argument to the function signature or leaving the code as-is.

Reply:
so, if someone calls `node_setup(wait_db_up=True)`, it will wait for DB to be up..
the part of `node_setup(wait_db_up=False)` not calling `clean_replacement_node_ip` could be a problem, but in any case, it shouldn't disturb either, as this variable in `scylla.yaml` is to be used on 1st boot, and afterwards isn't really used (@roydahan commented about it being a potential issue if some other action takes place afterwards).
So as a suggestion, I could add another function that will only `wait_db_up` that will only wait for DB to be up, and try to clean the `clean_replacement_node_ip`, but i'm not sure it is worth it, WDYT @bentsi ?

Reply:
moving this code to another function won't help. Let's keep it like this